### ISSUE TITLE
Open target link with external application, if it is not a markdown file

### ISSUE THREAD

**Issue by bkilinc** [marktext/marktext#2540]
### Describe your feature request
It should be possible to give links to files in local directory. These links should be opened with associated application (libreoffice etc.). This behaviour exists in typora.

### What problem does this feature solve? [optional]

Linking to local assets will be possible.

### Additional context [optional]

I briefly looked at source and I guess this should be possible modifiying some code in
https://github.com/marktext/marktext/blob/0fa6c33c44e92d0d1729d3d557d02ac7db87e73f/src/main/menu/actions/file.js
it is ''mt::format-link-click' event
```
ipcMain.on('mt::format-link-click', (e, { data, dirname }) => {
// ...
  let pathname = null
  if (path.isAbsolute(href) && isMarkdownFile(href)) {
    pathname = href
  } else if (!path.isAbsolute(href) && isMarkdownFile(path.join(dirname, href))) {
    pathname = path.join(dirname, href)
  }
  if (pathname) {
    const win = BrowserWindow.fromWebContents(e.sender)
    return openFileOrFolder(win, pathname)
  }
```
obviously, if target is not a Markdown file, it does nothing (pathname == null). so I think there will be an else after if(pathname) and it will handle resource. I will also try.

**Comment by bkilinc** [comment]
I solved this with following modification to the event. I will also prepare a pull request later. With this modification, it opens non-markdown files with associated application.
```
ipcMain.on('mt::format-link-click', (e, { data, dirname }) => {
  if (!data || (!data.href && !data.text)) {
    return
  }

  const href = data.href || data.text
  if (URL_REG.test(href)) {
    return shell.openExternal(href)
  }

  let pathname = path.isAbsolute(href) ? href : path.join(dirname, href)
  if (isMarkdownFile(pathname)) {
    const win = BrowserWindow.fromWebContents(e.sender)
    return openFileOrFolder(win, pathname)
  } else {
    return shell.openPath(pathname)
  }
})
```

**Comment by fxha** [comment]
Fixed by #2563.
